client/blocks/comment-detail/index.jsx

```diff
@@ -80,18 +81,40 @@ export class CommentDetail extends Component {
 
 	edit = () => noop;
 
+	getCommentObject = () => ( {
+		commentId: this.props.commentId,
+		isLiked: this.props.commentIsLiked,
+		postId: this.props.postId,
+		status: this.props.commentStatus,
+	} );
+
 	toggleApprove = () => {
-		const { commentId, commentStatus, postId, setCommentStatus } = this.props;
-		setCommentStatus( commentId, postId, 'approved' === commentStatus ? 'unapproved' : 'approved' );
+		const { commentStatus, setCommentStatus } = this.props;
+		const shouldPersist = 'approved' === commentStatus || 'unapproved' === commentStatus;
+
+		setCommentStatus( this.getCommentObject(), 'approved' === commentStatus ? 'unapproved' : 'approved', {
```

Comment:
I would also, suggest moving the next status on it's own line, like we do with shouldPersist. I was getting distracted with the extra arg after the ternary.